CMakeLists.txt

```diff
@@ -0,0 +1,212 @@
+    LEGO1/mxdsevent.cpp
+    LEGO1/mxdsfile.cpp
+    LEGO1/mxdsmediaaction.cpp
+    LEGO1/mxdsmultiaction.cpp
+    LEGO1/mxdsobject.cpp
+    LEGO1/mxdsobjectaction.cpp
+    LEGO1/mxdsparallelaction.cpp
+    LEGO1/mxdsselectaction.cpp
+    LEGO1/mxdsserialaction.cpp
+    LEGO1/mxdssound.cpp
+    LEGO1/mxdssource.cpp
+    LEGO1/mxdsstill.cpp
+    LEGO1/mxdssubscriber.cpp
+    LEGO1/mxentity.cpp
+    LEGO1/mxeventmanager.cpp
+    LEGO1/mxeventpresenter.cpp
+    LEGO1/mxflcpresenter.cpp
+    LEGO1/mxioinfo.cpp
+    LEGO1/mxloopingflcpresenter.cpp
+    LEGO1/mxloopingsmkpresenter.cpp
+    LEGO1/mxmediapresenter.cpp
+    LEGO1/mxmusicpresenter.cpp
+    LEGO1/mxnotificationmanager.cpp
+    LEGO1/mxomni.cpp
+    LEGO1/mxomnicreateflags.cpp
+    LEGO1/mxomnicreateparam.cpp
+    LEGO1/mxomnicreateparambase.cpp
+    LEGO1/mxpalette.cpp
+    LEGO1/mxpresenter.cpp
+    LEGO1/mxsmkpresenter.cpp
+    LEGO1/mxsoundmanager.cpp
+    LEGO1/mxsoundpresenter.cpp
+    LEGO1/mxstillpresenter.cpp
+    LEGO1/mxstreamer.cpp
+    LEGO1/mxstring.cpp
+    LEGO1/mxtimer.cpp
+    LEGO1/mxtransitionmanager.cpp
+    LEGO1/mxunknown100dc6b0.cpp
+    LEGO1/mxvideomanager.cpp
+    LEGO1/mxvideoparam.cpp
+    LEGO1/mxvideoparamflags.cpp
+    LEGO1/mxvideopresenter.cpp
+    LEGO1/mxwavepresenter.cpp
+    LEGO1/pizza.cpp
+    LEGO1/pizzamissionstate.cpp
+    LEGO1/pizzeria.cpp
+    LEGO1/pizzeriastate.cpp
+    LEGO1/police.cpp
+    LEGO1/policeentity.cpp
+    LEGO1/policestate.cpp
+    LEGO1/racecar.cpp
+    LEGO1/racestandsentity.cpp
+    LEGO1/racestate.cpp
+    LEGO1/radio.cpp
+    LEGO1/radiostate.cpp
+    LEGO1/registrationbook.cpp
+    LEGO1/score.cpp
+    LEGO1/scorestate.cpp
+    LEGO1/skateboard.cpp
+    LEGO1/towtrack.cpp
+    LEGO1/towtrackmissionstate.cpp
+  )
+
+  # Link libraries
+  target_link_libraries(lego1 PRIVATE ddraw dsound winmm)
+
+  # Make sure filenames are ALL CAPS
+  set_property(TARGET lego1 PROPERTY OUTPUT_NAME LEGO1)
+  set_property(TARGET lego1 PROPERTY SUFFIX ".DLL")
+endif()
+
+if (MSVC)
+  # These flags have been taken from the defaults for a Visual C++ 4.20 project (the compiler the
+  # game was originally built with) and tweaked slightly to produce more debugging info for reccmp.
+  # They ensure a recompilation that can be byte/instruction accurate to the original binaries.
+
+  target_compile_options(isle PRIVATE "/ML$<$<CONFIG:Debug>:d>")
+  target_compile_options(lego1 PRIVATE "/MT$<$<CONFIG:Debug>:d>")
+
+  set(CMAKE_CXX_FLAGS "/W3 /GX /D \"WIN32\" /D \"_WINDOWS\"")
+  set(CMAKE_CXX_FLAGS_DEBUG "/Gm /Zi /Od /D \"_DEBUG\"")
+  set(CMAKE_CXX_FLAGS_RELEASE "/O2 /D \"NDEBUG\"")
+  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/Zi /O2 /D \"NDEBUG\"")
+  set(CMAKE_CXX_FLAGS_MINSIZEREL "/Os /D \"NDEBUG\"")
+
+  set(CMAKE_EXE_LINKER_FLAGS "/machine:I386")
+  set(CMAKE_EXE_LINKER_FLAGS_DEBUG "/incremental:yes /debug")
+  set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/incremental:no")
+  set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "/incremental:no /debug")
```

Comment:
Great idea to make `RelWIthDebInfo` equal to `Release` apart from this `/debug` flag.